.evergreen/find-python3.sh

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env bash
+#
+# find-python3.sh
+#
+# Usage:
+#   . /path/to/find-python3.sh
+#
+# This file defines the following utility functions:
+#   - is_python3
+#   - is_venv_capable
+#   - is_virtualenv_capable
+#   - find_python3
+# These functions may be invoked from any working directory.
+
+# is_python3
+#
+# Usage:
+#   is_python3 python
+#   is_python3 /path/to/python
+#
+# Parameters:
+#   "$1": The name or path of the python binary to test.
+#
+# Return 0 (true) if the given argument "$1" is a Python 3 binary.
+# Return a non-zero value (false) otherwise.
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# to silence these messages.
+is_python3() (
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  # Binary to use, e.g. "python".
+  local -r bin="${1:?'is_python3 requires a name or path of a python binary to test'}"
+
+  # Binary must be executable.
+  command -V "$bin" &>/dev/null || return
+
+  # Reject binaries that do not support argument "-V".
+  "$bin" -V &>/dev/null || return
+
+  # Expect an output of the form: "Python x.y.z".
+  # Note: Python 2 binaries output to stderr rather than stdout.
+  local -r version_output="$("$bin" -V 2>&1)"
+
+  # For diagnostic purposes.
+  echo " - $bin: $version_output"
+
+  # "Python x.y.z" -> "x.y.z"
+  local -r version_str="$(perl -lne 'print $1 if m/.*([0-9]+\.[0-9]+\.[0-9]*)/' -- <(printf "%s" "$version_output"))"
+
+  # Evaluate 3.0.0 <= x.y.z.
+  sort -CV -- <(printf "%s\n%s\n" "3.0.0" "$version_str")
```

Comment:
FYI this logic can be implemented much simpler in python. For example:
```
if python -c "import sys; exit(sys.version_info[0] < 3)"; then
   echo "python 3"
else
   echo "python 2"
fi
```

this would avoid the dependency on "sort".

Reply:
[Verified](https://spruce.mongodb.com/version/63485ef132f417225c2244fb/tasks) this suggestion works as suggested. Thank you!

However, it does not completely remove the GNU Coreutils `sort` dependency as it is used below to prioritize newer MongoDB Toolchain and Python versions under `/opt`. It seems like macos-1012 is [circumstantially passing](https://evergreen.mongodb.com/lobster/evergreen/task/drivers_tools_test_find_python3_macos_1012_test_activate_kmstlsvenv_patch_2443440ca69bcdba825eb0b43f874006ae3b5bca_63485ef132f417225c2244fb_22_10_13_18_54_41/0/task#bookmarks=0%2C406&l=1&shareLine=245) due to system `python3` satisfying the requirements despite `sort` command failures when searching for binaries under `/opt`.